src/Node/ReadLine.purs

```diff
@@ -1,42 +1,86 @@
 -- | This module provides a binding to the Node `readline` API.
 
-module Node.ReadLine where
+module Node.ReadLine
+  ( Interface
+  , InterfaceOptions
+  , Completer
+  , LineHandler
+  , createInterface
+  , createInteractiveConsoleInterface
+  , noCompletion
+  , prompt
+  , setPrompt
+  , setLineHandler
+  , close
+  ) where
 
 import Prelude (return)
-
-import Control.Monad.Eff
-import Control.Monad.Eff.Console
+import Control.Monad.Eff (Eff)
+import Control.Monad.Eff.Console (CONSOLE)
+import Control.Monad.Eff.Exception (EXCEPTION)
+import Data.Maybe (Maybe(Just, Nothing), fromMaybe)
+import Node.Process (stdin, stdout)
+import Node.Stream (Readable, Writable)
 
 -- | A handle to a console interface.
 -- |
 -- | A handle can be created with the `createInterface` function.
 foreign import data Interface :: *
 
+foreign import createInterfaceImpl :: forall opts eff. { | opts } -> Eff ( | eff) Interface
```

Comment:
`{ | opts }` is syntax sugar for `Prim.Object ( | opts)` (which, I'm 99% sure, is the same as `Prim.Object opts`). You can also look at the kinds; `opts` will be `# *` here, whereas `{ | opts }` is `*`.